### ISSUE TITLE
Salt provisioner config "salt.version" doens't accept "2016.11.2"

### ISSUE THREAD

**Issue by RiccardoBoettcher** [mitchellh/vagrant#8299]
### Vagrant version
1.9.1

### Host operating system
Linux, Fedora 25

### Guest operating system
Windows 8.1

### Vagrantfile
```ruby
    config.vm.provision :salt do |salt|
        salt.version = "2016.11.2"
        salt.masterless = true
        salt.minion_config = "../salt/minion"
        salt.grains_config = "../salt/grains"
        salt.run_highstate = true
        salt.colorize = true
        salt.verbose = true
        salt.always_install = false
    end
```

### Debug output
```
==> devimage-win81-stage01: Running provisioner: salt...
Copying salt minion config to vm.
Copying salt grains config to vm.
Checking if C:\salt\salt-minion.bat is installed
C:\salt\salt-minion.bat was not found.
Checking if C:\salt\salt-call.bat is installed
C:\salt\salt-call.bat was not found.
Using Bootstrap Options:
Bootstrapping Salt... (this may take a while)
Service defaulting to run.

Downloading Salt minion installer Salt-Minion-2015.5.2-AMD64-Setup.exe
```

### Expected behavior
It should accept 2016.11.2 which is the most recent version now.

### Actual behavior
Falls back to built-in default value due to malformed reg-ex in 
https://github.com/mitchellh/vagrant/blob/master/plugins/provisioners/salt/bootstrap-salt.ps1
commit c8a18fd 
line 13
If ($version -notmatch "201[0-9]\.[0-9]\.[0-9](\-\d{1})?"){
  $version = '2015.5.2'
}

reg-ex should be 
If ($version -notmatch "20[0-9]{2}\.[0-9]{1,2}\.[0-9](\-\d{1})?"){
  $version = '2015.5.2'
}

### Steps to reproduce
Just spin up a box which uses the salt provisioner using a customized version

**Comment by romcab** [comment]
too bad, i just saw this post today. i tried different combinations of install_type and version yesterday. thanks for posting anyway.

**Comment by chrisroberts** [comment]
Fixed via #8428
